### PR TITLE
New download button

### PR DESCRIPTION
**Pull request by a-mabe**
- Added memento URI-M download button to summary page
- Modified "Revert" button function to increase load time
- Updated "X" button styling on the image grid tab

### CODE REVIEW

File: public/scripts/timeline.js [hunk 1]
```diff
@@ -1597,6 +1608,22 @@ function isValidDate(dateString)
...
+    $.each(object,function(index,obj){
+        if($(obj.event_html).attr("src").indexOf("notcaptured") < 0){
+            URIM[index] = obj.event_link;
+        }
+    });
+    var download = document.getElementById("downloadMementoURI");
+    download.href = "";
+    var data = URIM.toString();
+    data = data.replace(/,/g,"\n");
+    download.setAttribute('href', 'data:text/plaincharset=utf-8,' + encodeURIComponent(data));
```
Comment (ibnesayeed) [inline public/scripts/timeline.js]: You are missing a semicolon here:

```suggestion
    download.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(data));
```

File: public/scripts/timeline.js [hunk 2]
```diff
@@ -1597,6 +1608,22 @@ function isValidDate(dateString)
     return day > 0 && day <= monthLength[month];
 };
 
+function generateMementoURIList(object)
+{
+    var URIM = [];
+
+    $.each(object,function(index,obj){
+        if($(obj.event_html).attr("src").indexOf("notcaptured") < 0){
+            URIM[index] = obj.event_link;
```
Comment (ibnesayeed) [inline public/scripts/timeline.js]: If the `if` condition here is to filter some entries off, then the `URIM` array will be a sparse one in which skipped entries will be empty at their corresponding index. As a result, in the final output, there will be some empty lines. You can use `Array.push()` method here to eliminate empty entries in the first place like this:

```suggestion
            URIM.push(obj.event_link);
```

File: public/scripts/timeline.js [hunk 3]
```diff
@@ -1597,6 +1608,22 @@ function isValidDate(dateString)
...
+    var URIM = [];
+
+    $.each(object,function(index,obj){
+        if($(obj.event_html).attr("src").indexOf("notcaptured") < 0){
+            URIM[index] = obj.event_link;
+        }
+    });
+    var download = document.getElementById("downloadMementoURI");
+    download.href = "";
+    var data = URIM.toString();
```
Comment (ibnesayeed) [inline public/scripts/timeline.js]: You can join array entries on `\n` to eliminate the need of replacing commas with new lines later:

```suggestion
    var data = URIM.join('\n');
```

File: public/scripts/timeline.js [hunk 4]
```diff
@@ -1597,6 +1608,22 @@ function isValidDate(dateString)
...
+    data = data.replace(/,/g,"\n");
```
Comment (ibnesayeed) [inline public/scripts/timeline.js]: This line can now go away.

```suggestion
```

File: public/scripts/timeline.js [hunk 5]
```diff
@@ -1395,6 +1398,9 @@ $(function(){
     });
 
     $("#updateMementos").click(function(event){
+
+        let chosenMementos = JSON.parse(JSON.stringify(jsonObjRes));
```
Comment (ibnesayeed) [inline public/scripts/timeline.js]: Why are we first stringifying the object and then parsing it back?

Comment (a-mabe) [reply public/scripts/timeline.js]: It's creating a deep copy of the list of mementos so the user can delete mementos but also maintain the original list.

Comment (ibnesayeed) [reply public/scripts/timeline.js]: Makes sense, thanks!

File: public/scripts/timeline.js [hunk 6]
```diff
@@ -1597,6 +1608,21 @@ function isValidDate(dateString)
...
+{
+    var URIM = [];
+
+    $.each(object,function(index,obj){
+        if($(obj.event_html).attr("src").indexOf("notcaptured") < 0){
+            URIM.push(obj.event_link);
+        }
+    });
+    var download = document.getElementById("downloadMementoURI");
+    download.href = "";
```
Comment (ibnesayeed) [inline public/scripts/timeline.js]: Is this initialization/resetting of `href` attribute necessary here?

File: public/scripts/timeline.js [hunk 7]
```diff
@@ -1597,6 +1608,21 @@ function isValidDate(dateString)
...
+    var data = URIM.join('\n');
```
Comment (ibnesayeed) [inline public/scripts/timeline.js]: I think we can remove this unnecessary intermediate variable as it is not being reused anywhere other than in the immediate next line. We can perhaps place `URIM.join('\n')` inline in the `encodeURIComponent()` function below.

**Comment by ibnesayeed** [comment]
LGTM! Let me know when you feel it is well-tested and is ready to deploy.

File: public/scripts/timeline.js (hunk 6 above)
Comment (a-mabe) [reply public/scripts/timeline.js]: No, that line is not necessary. It has been removed.

File: public/scripts/timeline.js (hunk 7 above)
Comment (a-mabe) [reply public/scripts/timeline.js]: I moved `URIM.join('\n')` inline and tested the changes, everything is working as it should, thanks!

**Comment by a-mabe** [comment]
I've tested the changes and everything is working well, it's ready to deploy. Thank you!

**Comment by ibnesayeed** [comment]
The service is deployed now. You may want to test and ensure everything is working fine on our production site.

**Pull request merged**